process/block/poolsCleaner/miniBlocksPoolsCleaner.go

```diff
@@ -29,44 +34,42 @@ type mbInfo struct {
 
 // miniBlocksPoolsCleaner represents a pools cleaner that checks and cleans miniblocks which should not be in pool anymore
 type miniBlocksPoolsCleaner struct {
-	miniblocksPool   storage.Cacher
-	roundHandler     process.RoundHandler
-	shardCoordinator sharding.Coordinator
-
+	basePoolsCleaner
+	miniblocksPool         storage.Cacher
 	mutMapMiniBlocksRounds sync.RWMutex
 	mapMiniBlocksRounds    map[string]*mbInfo
-	cancelFunc             func()
 }
 
 // NewMiniBlocksPoolsCleaner will return a new miniblocks pools cleaner
-func NewMiniBlocksPoolsCleaner(
-	miniblocksPool storage.Cacher,
-	roundHandler process.RoundHandler,
-	shardCoordinator sharding.Coordinator,
-) (*miniBlocksPoolsCleaner, error) {
-
-	if check.IfNil(miniblocksPool) {
-		return nil, process.ErrNilMiniBlockPool
-	}
-	if check.IfNil(roundHandler) {
-		return nil, process.ErrNilRoundHandler
-	}
-	if check.IfNil(shardCoordinator) {
-		return nil, process.ErrNilShardCoordinator
+func NewMiniBlocksPoolsCleaner(args ArgMiniBlocksPoolsCleaner) (*miniBlocksPoolsCleaner, error) {
+	err := checkArgMiniBlocksPoolsCleaner(args)
+	if err != nil {
+		return nil, err
 	}
 
 	mbpc := miniBlocksPoolsCleaner{
-		miniblocksPool:   miniblocksPool,
-		roundHandler:     roundHandler,
-		shardCoordinator: shardCoordinator,
+		basePoolsCleaner:    newBasePoolsCleaner(args.ArgBasePoolsCleaner),
+		miniblocksPool:      args.MiniblocksPool,
+		mapMiniBlocksRounds: make(map[string]*mbInfo),
 	}
 
-	mbpc.mapMiniBlocksRounds = make(map[string]*mbInfo)
 	mbpc.miniblocksPool.RegisterHandler(mbpc.receivedMiniBlock, core.UniqueIdentifier())
 
 	return &mbpc, nil
 }
 
+func checkArgMiniBlocksPoolsCleaner(args ArgMiniBlocksPoolsCleaner) error {
+	err := checkBaseArgs(args.ArgBasePoolsCleaner)
+	if err != nil {
+		return err
+	}
+	if check.IfNil(args.MiniblocksPool) {
+		return process.ErrNilMiniBlockPool
+	}
+
+	return err
+}
+
 // StartCleaning actually starts the pools cleaning mechanism
 func (mbpc *miniBlocksPoolsCleaner) StartCleaning() {
 	var ctx context.Context
```

Comment:
as discussed, we might add a guard check here in order to not be able to call this method twice. The guard can be made up with a bool & mutex

Reply:
added